Update dependencies handling for the demo run

danielmellado opened the pull request:
This commit cleans up a all the dependency handling and automates
it. Besides, it also cleans the docs a bit and clarifies how to
run this.

Besides that, it also adds a github workflow for linting.

Signed-off-by: Daniel Mellado <dmellado@redhat.com>

maiqueb commented:
I'd rather not have random binaries shipped via this repo. 

I don't mind providing the same via pulling the src code (gitignoring it), and building the binaries from it. 

... I know, few extra steps, but, I wouldn't just run a random binary I got from a github repo. Especially when it requires root access ...

danielmellado committed:
Update dependencies handling for the demo run

This commit cleans up a all the dependency handling and automates
it. Besides, it also cleans the docs a bit and clarifies how to
run this.

Besides that, it also adds a github workflow for linting.

Signed-off-by: Daniel Mellado <dmellado@redhat.com>

maiqueb commented on .github/workflows/lint.yml:
```diff
@@ -0,0 +1,22 @@
+      - name: install tox
+        run: python -m pip install --upgrade tox virtualenv setuptools pip
+      - name: run linter
+        run: tox -ebashate
```
shouldn't we run the examples ?

danielmellado replied:
not really a need, as I don't think we should run the examples in a PR basis

maiqueb replied:
Why wouldn't we ? If we have something that assures a PR does not break the examples, shouldn't we rely on it ?

maiqueb commented on README.md:
```diff
@@ -1,15 +1,18 @@
 # fosdem2023-cni-unchained
+
```
all changes in this file are not "seen" by the render - i.e. the same page is rendered with / without these changes. 

I don't think they're really required.

danielmellado replied:
Just for the sake of markdownlint and my ease of mind xD

maiqueb commented on tox.ini:
```diff
@@ -0,0 +1,30 @@
+         \)                                      \
+         -print0 | xargs -0 bashate -v -iE006 -eE005,E042"
+
+[testenv:run_unlimited_bw_demo]
```
shouldn't we have a target for the other bandwidth scenario ?

danielmellado replied:
I'll hold onto that for when we're done with all the examples. I'll tackle whatever more needed ones in a follow-up PR.

maiqueb reviewed (changes requested):
Huge improvement on the install dependency front.

Just a couple of suggestions / questions.

Thanks !

maiqueb reviewed (approved):
OK, given https://github.com/maiqueb/fosdem2023-cni-unchained/pull/2#discussion_r1082700136, let's get this in, and add the tests / other config types afterwards.
